### PR TITLE
[python-package][dask] avoid warning for default n_jobs values

### LINKED ISSUE #6797: [python-package] [dask] Dask estimators raise an unavoidable warning
## Description

When training a model with the `lightgbm.dask` estimators, this warning is always emitted:

> /Users/runner/miniforge/envs/test-env/lib/python3.11/site-packages/lightgbm/dask.py:549: UserWarning: Parameter n_jobs will be ignored.
  _log_warning(f"Parameter {param_alias} will be ignored.")

Nothing in `lightgbm`'s public interface can suppress this, and it shows up even when using all default values of parameters. That's a little annoying and a little confusing.... it should be changed.

Specifically... for `num_threads` and its aliases, the warning should be not raised if the value is `-1` or `None`.

## Reproducible example

```python
import dask.array as da
import lightgbm as lgb
from distributed import Client, LocalCluster
from sklearn.datasets import make_blobs

X, y = make_blobs(n_samples=1000, n_features=50, centers=2)

cluster = LocalCluster()
client = Client(cluster)

dX = da.from_array(X, chunks=(100, 50))
dy = da.from_array(y, chunks=(100,))
dask_model = lgb.DaskLGBMClassifier(n_estimators=10)
dask_model.fit(dX, dy)
```

## Environment info

LightGBM version or commit hash: https://github.com/microsoft/LightGBM/commit/3654ecaaa1a33628dd2b1cc936c3f1cfd31079a4

Command(s) you used to install LightGBM

```shell
cmake -B build -S .
cmake --build build --target _lightgbm
sh build-python.sh install --precompile
```

## Additional Comments

This is coming from here:

https://github.com/microsoft/LightGBM/blob/3654ecaaa1a33628dd2b1cc936c3f1cfd31079a4/python-package/lightgbm/dask.py#L544-L550

I think because `n_jobs` is an alias for `num_threads`.

```python
from lightgbm.basic import _ConfigAliases
_ConfigAliases.get("num_threads")
# {'n_jobs', 'num_threads', 'num_thread', 'nthread', 'nthreads'}
```

And it's guaranteed to be present in `params` there, because it's in the signature of the estimators.

https://github.com/microsoft/LightGBM/blob/3654ecaaa1a33628dd2b1cc936c3f1cfd31079a4/python-package/lightgbm/dask.py#L1135

I noticed this in CI logs:

https://github.com/microsoft/LightGBM/actions/runs/12922031182/job/36037058867#step:3:6718

### PR DESCRIPTION
**Pull request by shiyu1994**
## Summary
- avoid warning in `lightgbm.dask._train()` when a `num_threads` alias is present with default sklearn-constructor values (`None` or `-1`)
- keep warning behavior unchanged for other explicit `num_threads` / `num_machines` values that are ignored
- add a Dask test covering both default `n_jobs` alias values (`-1`, `None`)

## Testing
- `python -m compileall -q python-package/lightgbm/dask.py tests/python_package_test/test_dask.py`
- targeted `test_dask.py` run could not be executed here because the available LightGBM test environment lacks `dask` / `distributed` and package install is blocked by network proxy

Fixes #6797

### CODE REVIEW

File: tests/python_package_test/test_dask.py [hunk 1]
```diff
@@ -1208,6 +1208,17 @@ def test_warns_and_continues_on_unrecognized_tree_learner(cluster):
         assert dask_regressor.fitted_
 
 
+@pytest.mark.parametrize("n_jobs", [-1, None])
+def test_does_not_warn_on_default_n_jobs_alias_values(cluster, n_jobs):
```
Comment (jameslamb) [inline tests/python_package_test/test_dask.py:1212]: This does not test the behavior described in #6797

1. doesn't actually check the log output to confirm a warning isn't raised
2. passes parameter `n_jobs` which is interesting, but **the issue says the warning is raised even when not passing any multithreading related parameters**

If we move forward with this PR, this needs to be updated to confirm that the issue is actually fixed.

**Review by jameslamb — CHANGES_REQUESTED** [review]
There is already an open PR attempting to close this, authored by (as far as I can tell) a human who put effort into it: #6987

I'd prefer to take that PR over this LLM-generated one.

It has been 8 months since they last interacted there though. I just asked them to respond. If we don't hear back soon, I'd be ok with closing that PR and pursuing this one. If we do that, please do see my notes about the tests.

**Comment by jameslamb** [comment]
I recall this was something you said you'd opened as an experiment with one of the "Claw" agents, but looks like you've abandoned it. We keep receiving other PRs for this issue, so going to close this.
